arch/arm/core/thread.c

```diff
@@ -91,10 +91,10 @@ void z_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	 * k_thread_create(). If K_THREAD_STACK_SIZEOF() is used, the
 	 * Guard size has already been take out of stackSize.
 	 */
-	stackEnd = pStackMem + stackSize - MPU_GUARD_ALIGN_AND_SIZE;
-#else
-	stackEnd = pStackMem + stackSize;
+	stackSize -= MPU_GUARD_ALIGN_AND_SIZE;
```

Comment:
Also, I am stressing this once more :) 
This workaround is only needed because we do not have a formal enforcement in the API of k_thread_create(), to use `size = K_THREAD_STACK_SIZEOF(K_THREAD_STACK_DEFINE(sym, <nominal size>))`